features/dd-sdk-android-trace/build.gradle.kts

```diff
@@ -39,6 +39,14 @@ plugins {
 
 android {
     defaultConfig {
+        buildFeatures {
+            buildConfig = true
+        }
+        buildConfigField(
+            "String",
+            "OPENTELEMETRY_API_VERSION_NAME",
+            "\"${libs.versions.openTelemetry.get()}\""
```

Comment:
maybe there is something from OpenTelemetry in the `META-INF` folder of the resulting package? this would help to avoid `BuildConfig` usage.

Also the value here may be wrong, because it is written for the version which was used to compile our library, but once final app is compiled, there is a chance that there is a newer OpenTelemetry version in the classpath, so it means that our telemetry will report the wrong version.

Additionally, in theory since this property is written at the our library compile time, we can know the version used just by looking on the SDK version. This means that we can set up logs processing pipeline, which will transform SDK version to the OpenTelemetry version used (but that would require to maintain this rule).

Reply:
Not necessarily. Customer may just have some 3rd party dependency which references newer OpenTelemetry API artifact (or maybe they have their own OpenTelemetry tooling as well). In this case overall dependency graph will be resolved to the newer version, it will be used in the application, but our telemetry will still report the older version (because it will be the one which was used to compiler our SDK, not the one which will be used to compile the application).